src/buildercore/threadbare/state.py

```diff
@@ -0,0 +1,130 @@
+import copy
+import contextlib
+
+CLEANUP_KEY = "_cleanup"
+
+
+class LockableDict(dict):
```

Comment:
in a context of concurrency this name is scary as it evokes locks as in semaphores. `FreezableDict`?

Reply:
another victim burnt by deadlocks ;) fair enough, I'll rename it